espnet/nets/pytorch_backend/e2e_asr_transducer.py

```diff
@@ -44,397 +49,360 @@ class E2E(ASRInterface, torch.nn.Module):
     Args:
         idim (int): dimension of inputs
         odim (int): dimension of outputs
-        args (namespace): argument Namespace containing options
+        args (Namespace): argument Namespace containing options
 
     """
 
     @staticmethod
     def add_arguments(parser):
-        """Extend arguments for transducer."""
-        group = parser.add_argument_group("transducer model setting")
-        # encoder
+        """Extend arguments for self-attention transducer.
+
+        Both Transformer and RNN modules are supported.
+        General options encapsulate both modules options.
+
+        """
+        group = parser.add_argument_group("transformer model setting")
+
+        # Encoder - general
         group.add_argument('--etype', default='blstmp', type=str,
-                           choices=['lstm', 'blstm', 'lstmp', 'blstmp', 'vgglstmp', 'vggblstmp', 'vgglstm', 'vggblstm',
-                                    'gru', 'bgru', 'grup', 'bgrup', 'vgggrup', 'vggbgrup', 'vgggru', 'vggbgru'],
+                           choices=['transformer', 'lstm', 'blstm', 'lstmp', 'blstmp',
```

Comment:
"transformer" or "self attention (sa)"?

Reply:
sorry, keep to use transformer to be consistent with the other names. please ignore the above.